src/Raven.Server/Web/System/AdminGcDebugHandler.cs

```diff
@@ -0,0 +1,424 @@
+                    writer.WritePropertyName(alloc.Type);
+                    writer.WriteStartObject();
+                    writer.WritePropertyName("Allocated");
+                    writer.WriteString(new Size((long)alloc.Allocations, SizeUnit.Bytes).ToString());
+
+                    var additionalLogging = alloc.Allocations != alloc.SmallObjectAllocations;
+                    if (additionalLogging)
+                    {
+                        writer.WriteComma();
+                        writer.WritePropertyName("AllocatedSmallObjects");
+                        writer.WriteString(new Size((long)alloc.SmallObjectAllocations, SizeUnit.Bytes).ToString());
+                        writer.WriteComma();
+                        writer.WritePropertyName("AllocatedLargeObjects");
+                        writer.WriteString(new Size((long)alloc.LargeObjectAllocations, SizeUnit.Bytes).ToString());
+                    }
+                    
+                    writer.WriteComma();
+                    writer.WritePropertyName("NumberOfAllocations");
+                    writer.WriteInteger(alloc.NumberOfAllocations);
+
+                    if (additionalLogging)
+                    {
+                        writer.WriteComma();
+                        writer.WritePropertyName("NumberOfSmallObjectAllocations");
+                        writer.WriteInteger(alloc.NumberOfSmallObjectAllocations);
+                        writer.WriteComma();
+                        writer.WritePropertyName("NumberOfLargeObjectAllocations");
+                        writer.WriteInteger(alloc.NumberOfLargeObjectAllocations);
+                    }
+
+                    writer.WriteEndObject();
+                }
+
+                writer.WriteEndObject();
+            }
+        }
+
+        [RavenAction("/admin/debug/memory/gc-events", "GET", AuthorizationStatus.Operator,
+            // intentionally not calling it debug endpoint because it isn't valid for us
+            // to do so in debug package (since we force a wait)
+            IsDebugInformationEndpoint = false)]
+        public async Task GcEvents()
+        {
+            var delay = GetIntValueQueryString("delay", required: false) ?? 10;
+
+            IReadOnlyCollection<GcEventsEventListener.Event> events;
+            using (var listener = new GcEventsEventListener())
+            {
+                await Task.Delay(TimeSpan.FromSeconds(delay));
+                events = listener.Events;
+            }
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                writer.WriteStartObject();
+
+                writer.WritePropertyName("Events");
+                writer.WriteStartArray();
+
+                var first = true;
+                foreach (var @event in events.OrderByDescending(x => x.DurationInMs))
```

Comment:
does it make sense? you are loosing order of the events here?

Reply:
Would want to see the ones that took most of the time. @haludi what do you think?

Reply:
I think we need to go this the chronical order here..

Reply:
I as well think that chronical order is more important

Reply:
If we are going to write all the events chronically sounds better.
Any way I can check later the duration of the events.
We can add also the top 3 or something like that to make it easier to understand if there is any issue of the GC duration.

If you don't order it, it is ordered by the end time, right?
I think it will be more convenient to order it by the start time.

Reply:
done